app/src/main/java/com/google/samples/apps/nowinandroid/navigation/NiaNavHost.kt

```diff
@@ -39,33 +39,32 @@ import com.google.samples.apps.nowinandroid.ui.NiaAppState
 @Composable
 fun NiaNavHost(
     appState: NiaAppState,
+    snackbarHostState: SnackbarHostState,
     modifier: Modifier = Modifier,
     startDestination: String = forYouNavigationRoute,
 ) {
     val navController = appState.navController
+    val interestsScrollState = rememberLazyListState()
     NavHost(
         navController = navController,
         startDestination = startDestination,
         modifier = modifier,
     ) {
-        // TODO: handle topic clicks from each top level destination
-        forYouScreen(onTopicClick = {})
-        bookmarksScreen(onTopicClick = {})
+        forYouScreen(onTopicClick = navController::navigateToInterests)
+        bookmarksScreen(
+            onTopicClick = navController::navigateToInterests,
+            snackbarHostState = snackbarHostState,
+        )
         searchScreen(
             onBackClick = navController::popBackStack,
-            onInterestsClick = { appState.navigateToTopLevelDestination(INTERESTS) },
-            onTopicClick = navController::navigateToTopic,
+            onInterestsClick = navController::navigateToInterests,
+            onTopicClick = navController::navigateToInterests,
         )
         interestsGraph(
-            onTopicClick = { topicId ->
-                navController.navigateToTopic(topicId)
-            },
-            nestedGraphs = {
-                topicScreen(
-                    onBackClick = navController::popBackStack,
-                    onTopicClick = {},
-                )
-            },
+            listState = interestsScrollState,
+            shouldShowTwoPane = appState.shouldShowTwoPane,
+            onTopicClick = navController::navigateToInterests,
+            onBackClick = navController::navigateToInterests,
```

Comment:
This is a bit strange, that a "back click" will add another destination to the backstack (instead of popping up)

Reply:
Good point, changed to `popBackStack`